ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/util/VlanUtils.java

```diff
@@ -35,93 +40,64 @@
  * @author marek.ryznar@amartus.com
  */
 public class VlanUtils {
-    private Set<Integer> usedVlans = new HashSet<>();
+
+    private final Set<Integer> usedVlans = new HashSet<>();
+    private Node node;
+    private final DataBroker dataBroker;
 
     private final static Set<Integer> possibleVlans = IntStream.range(1, 4094).boxed().collect(Collectors.toSet());
     private final static String VLAN_POOL_EXHAUSTED_ERROR_MESSAGE = "All VLAN IDs are in use. VLAN pool exhausted.";
 
     private final static Logger LOG = LoggerFactory.getLogger(VlanUtils.class);
 
-    public VlanUtils(List<NullAwareDatastoreGetter<Node>> nodes) throws ResourceNotAvailableException {
-        getAllVlanIDs(nodes);
+
+    public VlanUtils(DataBroker dataBroker, String nodeId ) throws ResourceNotAvailableException {
+        this.dataBroker = dataBroker;
+        try {
+            node = new NrpDao(dataBroker.newReadOnlyTransaction()).getNode(new Uuid(nodeId));
+            if (node == null) throw new ReadFailedException(null);
```

Comment:
why not to throw ResoureceNotAvailable directly?

Reply:
will be corrected